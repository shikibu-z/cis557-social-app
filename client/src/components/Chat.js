/* eslint-disable */
import React from 'react';
import '../stylesheets/Chat.css';
import {
  Link,
} from 'react-router-dom';

function Chat(props) {

  let { id, src, name, latestMessage, chosen } = props;

  // truncate too long message
  if (latestMessage.length > 30) {
    latestMessage = latestMessage.substring(0, 30) + "...";
  }

  return (
    <Link key={id} style={{ textDecoration: 'none', color: 'black' }} to={`/chat/${id}`}>
      {id === parseInt(chosen)
        ? <div className="chat-card" style={{ background: 'lightslategrey' }}>
          <img src={src} alt="" />
          <div className="chat-card-content">
            <h2 style={{ color: 'white' }}>{name}</h2>
            {/* <p style={{ color: 'white' }}>{latestMessage}</p> */}
          </div>
        </div>
        : <div className="chat-card">
          <img src={src} alt="" />
          <div className="chat-card-content">
            <h2>{name}</h2>
            {/* <p>{latestMessage}</p> */}
          </div>
        </div>
      }
    </Link>
  );
}

export default Chat;
