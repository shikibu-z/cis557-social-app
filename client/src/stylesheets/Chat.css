.chat-card {
  display: flex;
  flex-direction: row;
  align-self: center;
  background-color: white;
  width: 90%;
  height: 90px;
  border-radius: 20px;
  box-shadow: 0px 5px 5px grey;
  margin: 0% 3% 4% 5%;
}

.chat-card img {
  width:60px;
  height:60px;
  border-radius: 50%;
  overflow:hidden;
  margin: auto 4% auto 4%;
}

.chat-card-content {
  display: felx;
  flex-direction: column;
  margin: 3% auto auto 0%;
}

.chat-card-content h2 {
  margin: auto 0% auto;
  font-size: 30px;
  font-weight:400;
  color: #595959;
}