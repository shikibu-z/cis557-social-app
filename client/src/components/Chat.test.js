/*eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Chat from './Chat';

test('Chat matches snapshot', () => {
  const component = renderer.create(<BrowserRouter><Chat latestMessage="good" /></BrowserRouter>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
