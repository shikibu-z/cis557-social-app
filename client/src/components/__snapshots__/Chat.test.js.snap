// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat matches snapshot 1`] = `
<a
  href="/chat/undefined"
  onClick={[Function]}
  style={
    Object {
      "color": "black",
      "textDecoration": "none",
    }
  }
>
  <div
    className="chat-card"
  >
    <img
      alt=""
    />
    <div
      className="chat-card-content"
    >
      <h2 />
    </div>
  </div>
</a>
`;
